swap/peer.go

```diff
@@ -69,10 +69,10 @@ func (sp *Peer) handleMsg(ctx context.Context, msg interface{}) error {
 // TODO: this should not be blocking
 func (sp *Peer) handleEmitChequeMsg(ctx context.Context, msg *EmitChequeMsg) error {
 	cheque := msg.Cheque
-	log.Debug("received emit cheque message from peer", "peer", sp.ID().String())
+	log.Info("received cheque from peer, attempting to submit and cash", "peer", sp.ID().String())
 	actualAmount, err := sp.processAndVerifyCheque(cheque)
 	if err != nil {
-		log.Error("invalid cheque from peer", "peer", sp.ID().String(), "error", err.Error())
+		log.Warn("invalid cheque from peer", "peer", sp.ID().String(), "error", err.Error())
```

Comment:
We shouldn't log an error and then return it. We can:
- log in case there is no error (with the appropriate level).
- return the error, then log it on the upper level that handles it.